cmd/frontend/graphqlbackend/survey_response.go

```diff
@@ -119,3 +120,58 @@ func (r *schemaResolver) SubmitSurvey(ctx context.Context, args *struct {
 
 	return &EmptyResponse{}, nil
 }
+
+// FeedbackSubmissionInput contains a happiness feedback response.
+type HappinessFeedbackSubmissionInput struct {
+	// Score is the user's happiness rating, from 1-4.
+	Score int32
+	// Feedback is the answer to "What's going well? What could be better?".
+	Feedback *string
+	// The URL that the happiness feedback was submitted from
+	CurrentURL string
+}
+
+type happinessFeedbackSubmissionForHubSpot struct {
+	Email      *string `url:"email"`
+	Score      int32   `url:"happiness_score"`
+	Feedback   *string `url:"happiness_feedback"`
+	CurrentURL string  `url:"happiness_current_url"`
+	SiteID     string  `url:"site_id"`
+}
+
+// SubmitHappinessFeedback records a new happiness feedback response by the current user.
+func (r *schemaResolver) SubmitHappinessFeedback(ctx context.Context, args *struct {
+	Input *HappinessFeedbackSubmissionInput
+}) (*EmptyResponse, error) {
+	var email *string
+
+	// If we are on Sourcegraph.com, we want to capture the email of the user
+	if envvar.SourcegraphDotComMode() {
+		actor := actor.FromContext(ctx)
+
+		// If user is authenticated, use their uid and set the email field.
+		if actor.IsAuthenticated() {
+			e, _, err := database.GlobalUserEmails.GetPrimaryEmail(ctx, actor.UID)
```

Comment:
I think using the global databases is discouraged now, so this would be using a local conn :)

```suggestion
			e, _, err := database.UserEmails(r.db).GetPrimaryEmail(ctx, actor.UID)
```